backend/dataall/modules/datasets/services/dataset_service.py

```diff
@@ -53,33 +53,22 @@
 
 class DatasetServiceInterface(ABC):
     @staticmethod
+    @abstractmethod
     def check_before_delete(session, uri, **kwargs) -> bool:
         """Abstract method to be implemented by dependent modules that want to add checks before deletion for dataset objects"""
-        return True
+        raise NotImplementedError('Method check_before_delete is not implemented')
```

Comment:
just an ellipsis here will do to avoid the visual clutter, python will throw because of the decorator

check this out...
```python
from abc import abstractmethod, ABC

import pytest


class FooABC(ABC):
    @staticmethod
    @abstractmethod
    def abc():
        ...


class FooImplMissing(FooABC):
    ...


def test_fooabc_insta_error():
    with pytest.raises(TypeError, match="Can't instantiate abstract class FooABC with abstract method abc"):
        FooABC()


def test_fooimpl_missing_error():
    with pytest.raises(TypeError, match="Can't instantiate abstract class FooImplMissing with abstract method abc"):
        FooImplMissing()
```